File: src/test/java/korzin/io/algorithms/BalancedBracketsTest.java
package korzin.io.algorithms;

import org.junit.Test;

import static org.junit.Assert.assertFalse;
import static org.junit.Assert.assertTrue;

public class BalancedBracketsTest {

    private BalancedBrackets algorithm = new BalancedBrackets();

    @Test
    public void hasBalancedBracketsTest() {

        assertTrue(algorithm.hasBalancedBrackets("(vfddf[]fvv)vf[]fvvdfdfv()"));
        assertTrue(algorithm.hasBalancedBrackets("((([scsccs([6576887564])354dscsdscs]sdc))sdc())"));
        assertFalse(algorithm.hasBalancedBrackets("(4356[]]()"));
        assertFalse(algorithm.hasBalancedBrackets("]["));
    }
}

File: src/test/java/korzin/io/algorithms/BalancedBracketsTest.kt
package korzin.io.algorithms

import org.junit.Assert.assertFalse
import org.junit.Assert.assertTrue
import org.junit.Test

class BalancedBracketsTest {

    @Test
    fun isBracketBalanced() {

        val algorithm = BalancedBrackets()
        assertTrue(algorithm.isBracketBalanced("([sc])s/ugf43&c[sd 1+565]()sdcsc"))
        assertTrue(algorithm.isBracketBalanced("((([scsccs([6576887564])354dscsdscs]sdc))sdc())"))
        assertFalse(algorithm.isBracketBalanced("(4356[]]()"))
        assertFalse(algorithm.isBracketBalanced("]["))
    }

}